From da768f6c94fc8af2d4faccd3a4d28d0b54576432 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Wed, 16 Sep 2020 14:36:46 +0530
Subject: [PATCH 0861/1921] perf reg: Fix compilation error with GCC 10
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Due to NULL being returned in default case GCC 10
is throwing below error. This is a temporary fix
for the S2 release.

In file included from /home/sgoutham/tools/marvell-tools-1010.0/aarch64-marvell-linux-gnu/sys-root/usr/include/stdio.h:867,
                 from /home/sgoutham/WORK/new/sunil/combined/5.4/linux/tools/perf/lib/include/perf/cpumap.h:6,
                 from util/session.c:13:
In function ‘printf’,
    inlined from ‘regs_dump__printf’ at util/session.c:1098:3,
    inlined from ‘regs__printf’ at util/session.c:1126:2:
/home/sgoutham/tools/marvell-tools-1010.0/aarch64-marvell-linux-gnu/sys-root/usr/include/bits/stdio2.h:107:10: error: ‘%-5s’ directive argument is null [-Werror=format-overflow=]
  107 |   return __printf_chk (__USE_FORTIFY_LEVEL - 1, __fmt, __va_arg_pack ());
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Change-Id: Ib04ab83cccb00cf6439a75a5d0ef64973dffd02b
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/35922
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
(cherry picked from commit afff581bdec797a523eb6e9765a6fbe7c14eb79d)
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/35967
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 tools/perf/arch/arm64/include/perf_regs.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/perf/arch/arm64/include/perf_regs.h b/tools/perf/arch/arm64/include/perf_regs.h
index baaa5e64a3fb..d9603bfde834 100644
--- a/tools/perf/arch/arm64/include/perf_regs.h
+++ b/tools/perf/arch/arm64/include/perf_regs.h
@@ -85,7 +85,7 @@ static inline const char *perf_reg_name(int id)
 	case PERF_REG_ARM64_PC:
 		return "pc";
 	default:
-		return NULL;
+		return "NULL";
 	}
 
 	return NULL;
-- 
2.31.1

